### PR TITLE
148 - support also .zappr.yaml beside .zappr.yml

### PR DESCRIPTION
**Pull request by spaudanjo**
fixes: #148

### CODE REVIEW

File: server/service/GithubService.js [hunk 1]
```diff
@@ -130,27 +131,28 @@ export default class GithubService {
     this.fetchPath('POST', path, payload, accessToken)
   }
 
+
   async readZapprFile(user, repo, accessToken) {
```
Comment (prayerslayer) [inline server/service/GithubService.js]: While we're at it we could rename the `user` parameter to `owner` for reduced mental overhead.

File: server/service/GithubService.js [hunk 2]
```diff
@@ -130,27 +131,28 @@ export default class GithubService {
...
-    // fetch file info
-    const path = PATHS.ZAPPR_FILE_REPO.replace('${owner}', user).replace('${repo}', repo)
+    const repoContentUrl = API_URL_TEMPLATES.REPO_CONTENT.replace('${owner}', user).replace('${repo}', repo)
+    const validZapprFileUrls = nconf.get('VALID_ZAPPR_FILE_PATHS')
```
Comment (prayerslayer) [inline server/service/GithubService.js]: The `VALID_ZAPPR_FILE_PATHS` we can also get just once?

File: server/service/GithubService.js [hunk 3]
```diff
@@ -130,27 +131,28 @@ export default class GithubService {
...
+                                    .map(zapprFilePath => path.join(repoContentUrl, zapprFilePath))
+
+    const zapprFileRequests = validZapprFileUrls.map(zapprFileUrl => this.fetchPath('GET', zapprFileUrl, null, accessToken))
+
     try {
-      let {content} = await this.fetchPath('GET', path, null, accessToken)
-      // short circuit if there is no such file
-      if (!content) {
+        const resp = await promiseFirst(zapprFileRequests)
+        const file = Buffer(resp.content, 'base64').toString('utf8')
```
Comment (prayerslayer) [inline server/service/GithubService.js]: `new Buffer`

File: server/service/GithubService.js (hunk 2 above)
Comment (spaudanjo) [inline server/service/GithubService.js]: owner is the resource owner, user the collaborator

File: server/service/GithubService.js (hunk 1 above)
Comment (spaudanjo) [inline server/service/GithubService.js]: owner is the resource owner, user the collaborator

**Comment by prayerslayer** [comment]
:+1:

**Pull request merged**